tests/Aspire.Hosting.Milvus.Tests/MilvusFunctionalTests.cs

```diff
@@ -0,0 +1,368 @@
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{milvus.Resource.Name}"] = await milvus.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None)
+        });
+
+        hb.AddMilvusClient(milvus.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        var milvusClient = host.Services.GetRequiredService<MilvusClient>();
+
+        string collectionName = "book";
+        var collection = await milvusClient.CreateCollectionAsync(
+                collectionName,
+                new[] {
+                FieldSchema.Create<long>("book_id", isPrimaryKey:true),
+                FieldSchema.Create<long>("word_count"),
+                FieldSchema.CreateVarchar("book_name", 256),
+                FieldSchema.CreateFloatVector("book_intro", 2)
+                }
+            );
+
+        var collections = await milvusClient.ListCollectionsAsync();
+
+        Assert.Single(collections, c => c.Name == collectionName);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyMilvusDatabaseResource()
+    {
+        var builder = CreateDistributedApplicationBuilder();
+
+        builder.Configuration["Parameters:apikey"] = MilvusToken;
+        var apiKey = builder.AddParameter("apikey");
+        var milvus = builder.AddMilvus("milvus", apiKey: apiKey);
+        var db = milvus.AddDatabase("milvusdb", "db1");
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(CancellationToken.None)
+        });
+
+        hb.AddMilvusClient(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        var milvusClient = host.Services.GetRequiredService<MilvusClient>();
+        await milvusClient.CreateDatabaseAsync("db1");
+        string collectionName = "book";
+        var collection = await milvusClient.CreateCollectionAsync(
+                collectionName,
+                new[] {
+                FieldSchema.Create<long>("book_id", isPrimaryKey:true),
+                FieldSchema.Create<long>("word_count"),
+                FieldSchema.CreateVarchar("book_name", 256),
+                FieldSchema.CreateFloatVector("book_intro", 2)
+                }
+            );
+
+        var collections = await milvusClient.ListCollectionsAsync();
+
+        Assert.Single(collections, c => c.Name == collectionName);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task WithDataVolumeShouldPersistStateBetweenUsages()
```

Comment:
Can we make this and the DataBindMount tests into a Theory? See https://github.com/dotnet/aspire/pull/4862 and https://github.com/dotnet/aspire/pull/4879 for examples.